agera/src/main/java/com/google/android/agera/Observables.java

```diff
@@ -50,16 +50,16 @@ public static Observable compositeObservable(@NonNull final Observable... observ
   static Observable compositeObservable(final int shortestUpdateWindowMillis,
       @NonNull final Observable... observables) {
     if (observables.length == 0) {
-      return new CompositeObservable(0);
+      return new CompositeObservable(shortestUpdateWindowMillis);
     }
 
     if (observables.length == 1) {
       final Observable singleObservable = observables[0];
       if (singleObservable instanceof CompositeObservable) {
-        return new CompositeObservable(0,
+        return new CompositeObservable(shortestUpdateWindowMillis,
```

Comment:
you probably need to check `&& singleObservable.shortestUpdateWindowMillis == 0` here before directly reusing the source observables in the singleObservable.